src/coreclr/pal/src/CMakeLists.txt

```diff
@@ -361,3 +361,9 @@ endif(FEATURE_EVENT_TRACE)
 
 # Install the static PAL library for VS
 install_clr (TARGETS coreclrpal DESTINATIONS lib)
+
+# Install the static PAL library for the runtime pack
+# The libcoreclrpal.a is linked using -Wl,--whole-archive as a separate static library
```

Comment:
The comment about `whole-archive` says: 

https://github.com/search?q=repo%3Adotnet%2Fruntime%20%22force%20all%20PAL%20objects%20to%20be%20included%20so%20all%20exports%20are%20available%22&type=code

I assume that these exports are needed in libcoreclr.so to satisfy imports from some other libraries (mayeb libclrjit?)

The `whole-archive` should not be needed when we are linking the whole thing together. It is unnecessary de-optimization when we are linking the whole thing together.

Reply:
My understanding is that libcoreclrpal.a provides platform‐initialization or OS‐abstraction code without direct external symbol dependencies. If it’s merged into coreclr_static.a, the native linker may strip away these symbols. To avoid forcing a whole‐archive on coreclr_static.a, link separately libcoreclrpal.a in --whole-archive:
```
add_library(
    monodroid
    SHARED
    monodroid-coreclr.c
)

target_link_libraries(
    monodroid
    PRIVATE
libcoreclr_static.a
-Wl,--whole-archive
libcoreclrpal.a
 -Wl,--no-whole-archive
)
```

I don’t have full clarity on where/how it’s used in the SDK.

/cc: @grendello

Reply:
> My understanding is that libcoreclrpal.a provides platform‐initialization or OS‐abstraction code

None of this is exposed outside coreclr. This is all internal coreclr implementation details. Nothing in SDK should depend on this.

> If it’s merged into coreclr_static.a, the native linker may strip away these symbols

Have you seen any actual failures that this causes?

Reply:
No. The functional test only links against coreclr_static.a, and the runtime is initialized correctly. @grendello could you please provide more context here?

Reply:
At this point, not really. We know there's a crash somewhere in the statically-linked runtime (the stack trace wouldn't give us any location, alas) and we know that the crash was avoided by linking `libcoreclrpal.a` as a whole. However, that moved the crash to a different location, so it might be entirely co-incidental. I assume that this archive is linked by the "normal" CoreCLR build as a whole because parts of it are called dynamically by the managed land (ecalls or p/invokes) and, thus, may not be referenced directly by the native linker and would be removed by it otherwise. If this is the case, then static linking of the Android runtime should use the same approach.